OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -70,7 +70,7 @@ public AttackBase(Actor self, AttackBaseInfo info)
 
 		protected virtual bool CanAttack(Actor self, Target target)
 		{
-			if (!self.IsInWorld || IsTraitDisabled)
+			if (!self.IsInWorld || self.IsDisabled() || IsTraitDisabled)
```

Comment:
`if (!self.IsInWorld || IsTraitDisabled || self.IsDisabled())`  for bonus cookie points.